src/components/MediaViewer/MediaViewer.tsx

```diff
@@ -351,500 +366,615 @@ const MediaViewer = ({ resource }: { resource: CloudinaryResource }) => {
-          src={resource.public_id}
-          alt={`Image ${resource.public_id}`}
-          style={imgStyles}
-          version={version}
-          placeholderStyle="dark"
-          {...transformations}
-        />
+            {!isTrash && (
+              <li>
+                {process.env.NEXT_PUBLIC_PHOTOBOX_MODE !== "read-only" && (
+                  <Button
+                    variant="ghost"
+                    className="text-white"
+                    onClick={() => setDeletion({ state: "confirm" })}
+                  >
+                    <Trash2 className="h-6 w-6" />
+                    <span className="sr-only">Delete</span>
+                  </Button>
+                )}
+                {process.env.NEXT_PUBLIC_PHOTOBOX_MODE === "read-only" && (
+                  <TooltipProvider delayDuration={200}>
+                    <Tooltip>
+                      <TooltipTrigger
+                        className={`${buttonVariants({ variant: "ghost" })} text-zinc-500 hover:text-zinc-500 bg-transparent hover:bg-transparent`}
+                        aria-label="Deleting is disabled"
+                      >
+                        <Trash2 className="h-6 w-6" />
+                        <span className="sr-only">Delete</span>
+                      </TooltipTrigger>
+                      <TooltipContent>
+                        <p>Deleting is Disabled</p>
+                      </TooltipContent>
+                    </Tooltip>
+                  </TooltipProvider>
+                )}
+              </li>
+            )}
+            {isTrash && (
+              <li>
+                <Button
+                  variant="ghost"
+                  className="text-white"
+                  onClick={handleOnRestore}
+                >
+                  <History className="h-6 w-6" />
+                  <span className="sr-only">Restore</span>
+                </Button>
+              </li>
+            )}
+          </ul>
+        </Container>
+
+        {/** Asset viewer */}
+
+        <div className="relative flex justify-center items-center align-center w-full h-full">
+          <CldImage
+            key={`${JSON.stringify(transformations)}-${version}`}
+            className="object-contain"
+            width={resource.width}
+            height={resource.height}
+            src={resource.public_id}
+            alt={`Image ${resource.public_id}`}
+            style={imgStyles}
+            version={version}
+            placeholderStyle="dark"
+            {...transformations}
+          />
+        </div>
       </div>
-    </div>
-  )
+
+      {/** Confirmation dialog to display if there are unsaved edits */}
+
+      <Dialog open={showConfirmDialog}>
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>Save Changes?</DialogTitle>
+          </DialogHeader>
+          <p>
+            You have unsaved changes that will be lost. Do you still want to
+            navigate back to the main gallery?
```

Comment:
I'm not sure what you wanted for the copy, so feel free to suggest something different.

Reply:
looks good to me!